hdkeychain/extendedkey.go

```diff
@@ -295,9 +297,9 @@ func (k *ExtendedKey) Child(i uint32) (*ExtendedKey, error) {
 		// derive the final child key.
 		//
 		// childKey = serP(point(parse256(Il)) + parentKey)
-		childX, childY := chainec.Secp256k1.Add(ilx, ily, pubKey.GetX(),
+		childX, childY := curve.Add(ilx, ily, pubKey.GetX(),
 			pubKey.GetY())
-		pk := chainec.Secp256k1.NewPublicKey(childX, childY)
+		pk := secp256k1.NewPublicKey(childX, childY)
```

Comment:
Once again, please use the struct directly to avoid the GC overhead in crypto and to more closely match upstream.
```Go
pk := secp256k1.PublicKey{Curve: secp256k1.S256(), X: childX, Y: childY}
```